test/Test/Oscoin/Crypto/PubKey.hs

```diff
@@ -0,0 +1,59 @@
+module Test.Oscoin.Crypto.PubKey
+    ( tests
+    ) where
+
+import           Oscoin.Prelude
+
+import qualified Oscoin.Crypto.Hash as Crypto
+import qualified Oscoin.Crypto.PubKey as Crypto
+
+import           Oscoin.Test.Crypto
+import           Oscoin.Test.Crypto.PubKey.Arbitrary
+                 (arbitraryKeyPair, arbitrarySigned)
+import           Oscoin.Test.Util (condensed)
+
+import           Test.QuickCheck.Extended
+import           Test.QuickCheck.Instances ()
+import           Test.Tasty
+import           Test.Tasty.QuickCheck hiding ((===))
+
+import qualified Codec.Serialise as CBOR
+import qualified Data.Aeson as Aeson
+import           Data.ByteArray.Orphans ()
+
+tests :: forall c. Dict (IsCrypto c) -> TestTree
+tests d = testGroup "Oscoin.Crypto.PubKey"
+    [ testProperty "prop_jsonRoundtripSigned" (prop_jsonRoundtripSigned d)
+    , testProperty "prop_serialiseRoundtripSigned" (prop_serialiseRoundtripSigned d)
+    , testProperty "prop_jsonRoundtripPublicKey" (prop_jsonRoundtripPublicKey d)
+    , testProperty "prop_serialiseRoundtripPublicKey" (prop_serialiseRoundtripPublicKey d)
+    , testProperty "prop_hashSigned" (prop_hashSigned d)
+    ]
+
+prop_jsonRoundtripSigned :: forall c. Dict (IsCrypto c) -> Property
+prop_jsonRoundtripSigned Dict =
```

Comment:
If we're not using auto-discovery-template-magic, do we need the funky prefix vs. standard camel-case?

Reply:
No, we don’t really need this.This is just for consistency with the tests from the P2P layer.

Reply:
Well, I think we _want_ to spare us the typing down the road (hoping to get rid of `tasty` eventually). Module-local TH magic (like this: https://hackage.haskell.org/package/hedgehog-0.6/docs/Hedgehog.html#v:discover) is pretty acceptable for this imho, whereas the compiler plugin approach est un peu beaucoup trop.